### PR TITLE
Configuring providers guide

### LINKED ISSUE #10400: Configuring providers guide
### Description

Provide a guide explaining the core concepts around installing, uninstalling, enabling, disabling, and configuring providers.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #10400

### CODE REVIEW

**Comment by thomasdarimont** [comment]
LGTM

**Comment by tanadeau** [comment]
One of the questions I had in one of the GitHub discussions was if and how you could pass SPI config via environment variables, and the answer was yes via `KC_SPI_...`. Since this mentions the config format, perhaps a mention of that would be helpful.

**Comment by dasniko** [comment]
Sounds good to me, but adding a hint for the env var format `KC_SPI_...` like @tanadeau already mentioned, would be good.

Additionally, perhaps an admonition or something similar to make clear, that just setting config values for a provider is done with runtime config and setting/enabling/disabling a provider is done at buildtime would make things more obviously to customers, as I already see confusion at users with the buildtime/runtime configs now used in Keycloak.

File: docs/guides/src/main/server/configuration-provider.adoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+After reading this guide, you should be able to use the concepts and the steps herein explained to install, uninstall, enable, disable, and configure
+any provider, including those you have implemented to extend the server capabilities in order to better fulfill your requirements.
+
+== Configuration option format
+
+Providers can be configured by using a specific configuration format. The format consists of:
+
+```
+spi-<spi>-<provider-id>-<property>=<value>
```
Comment (stianst) [inline docs/guides/src/main/server/configuration-provider.adoc:21]:
```suggestion
spi-<spi-id>-<provider-id>-<property>=<value>
```

File: docs/guides/src/main/server/configuration-provider.adoc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+The `<spi>` is the name of the SPI you want to configure.
```
Comment (stianst) [inline docs/guides/src/main/server/configuration-provider.adoc:24]:
```suggestion
The `<spi-id>` is the name of the SPI you want to configure.
```

File: docs/guides/src/main/server/configuration-provider.adoc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+```
+spi-connections-http-client-default-connection-pool-size=10
+```
+
+== Setting a provider configuration option
+
+The configuration options you set to a provider can be provided when starting the server. The values are available to the providers
+when at the init phase of the provider lifecycle.
+
+If using the CLI, you should be able to set these options as follows:
```
Comment (stianst) [inline docs/guides/src/main/server/configuration-provider.adoc:43]: _The values are available to the providers
when at the init phase of the provider lifecycle_ -- I had to read this a few times to understand what you meant here. I'd leave this out as it's more a concern when writing custom providers, and should rather be mentioned in the extending the server guide than here.

Could probably also simplify this whole section to something like:

```suggestion
Provider configuration options are provided when starting the server, as shown in the following example:
```

File: docs/guides/src/main/server/configuration-provider.adoc [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+.Setting the `connection-pool-size` for the `default` provider of the `connections-http-client` SPI
+<@kc.start parameters="--spi-connections-http-client-default-connection-pool-size=10"/>
+
+== Configuring a default provider
```
Comment (stianst) [inline docs/guides/src/main/server/configuration-provider.adoc:48]:
```suggestion
== Configuring the default provider
```

**Comment by pedroigor** [comment]
@danielFesenmeyer @tanadeau The different configuration sources are covered in the [Configuration Guide](https://www.keycloak.org/server/configuration).

Not sure if we should explain here how to use env vars. Would make sense to include a link to the configuration guide in the `References` section?

**Comment by pedroigor** [comment]
> Additionally, perhaps an admonition or something similar to make clear, that just setting config values for a provider is done with runtime config and setting/enabling/disabling a provider is done at buildtime would make things more obviously to customers, as I already see confusion at users with the buildtime/runtime configs now used in Keycloak.

@dasniko I tried to make it obvious by providing examples using the appropriate command as well as a bit of text explaining the reason. But if you are asking is because it is not yet clear. Do you have any suggestions on how to make this clear?

**Comment by dasniko** [comment]
> I tried to make it obvious by providing examples using the appropriate command as well as a bit of text explaining the reason. But if you are asking is because it is not yet clear. Do you have any suggestions on how to make this clear?

@pedroigor For me, it's clear. But I notice some difficutlies for users in other forums to cope with now buildtime and runtime configs, as this is a new approach. Nothing special for the SPI config, but general config approach. Even when using the new config guide, people don't see that there are build and run configs. Sad, but true.

**Comment by pedroigor** [comment]
@dasniko Yes, you are right. And we are trying since the beginning to make the build/runtime concepts less painful. Not an easy task, but I hope we can continuously improve the UX and make the `build` step more natural when configuring the server.

**Comment by tanadeau** [comment]
@pedroigor, just adding an environment example with the example you already have would be very helpful.

```
KC_SPI_CONNECTIONS_HTTP_CLIENT_DEFAULT_CONNECTION_POOL_SIZE
```

**Comment by pedroigor** [comment]
> @pedroigor, just adding an environment example with the example you already have would be very helpful.
> 
> ```
> KC_SPI_CONNECTIONS_HTTP_CLIENT_DEFAULT_CONNECTION_POOL_SIZE
> ```

I see. As I mentioned before, I would like to avoid duplicating information in guides. We kinda have that covered in the configuration guide. Formatting options to their corresponding env format should be trivial if you read that guide?

**Review by stianst — APPROVED** [review]

**Comment by pedroigor** [comment]
@dasniko Could you please give your thoughts about https://github.com/keycloak/keycloak/discussions/10451?

**Pull request merged**